apps/src/javalab/JavalabEditorHeader.jsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import {openEditorDialog} from './javalabRedux';
+import msg from '@cdo/locale';
+import javalabMsg from '@cdo/javalab/locale';
+import PaneHeader, {
+  PaneSection,
+  PaneButton
+} from '@cdo/apps/templates/PaneHeader';
+import {DisplayTheme} from './DisplayTheme';
+import Backpack from './Backpack';
+import {JavalabEditorDialog} from './JavalabEditorDialogManager';
+import ProjectTemplateWorkspaceIcon from '../templates/ProjectTemplateWorkspaceIcon';
+
+/**
+ * Renders the header portion of the Java Lab editor, consisting of the New File,
+ * Backpack, Version History, and Commit Code buttons, as well as header text.
+ */
+function JavalabEditorHeader({
+  onBackpackImportFile,
+  openEditorDialog,
+  isReadOnlyWorkspace,
+  backpackEnabled,
+  displayTheme,
+  showProjectTemplateWorkspaceIcon
+}) {
+  const editorHeaderText = isReadOnlyWorkspace
+    ? msg.readonlyWorkspaceHeader()
+    : javalabMsg.editor();
+
+  return (
+    <PaneHeader hasFocus>
+      <PaneButton
+        id="javalab-editor-create-file"
+        iconClass="fa fa-plus-circle"
+        onClick={() => openEditorDialog(JavalabEditorDialog.CREATE_FILE)}
+        headerHasFocus
+        isRtl={false}
+        label={javalabMsg.newFile()}
+        leftJustified
+        isDisabled={isReadOnlyWorkspace}
+      />
+      {backpackEnabled && (
+        <PaneSection style={styles.backpackSection}>
+          <Backpack
```

Comment:
Not sure if this is worth noting somewhere, but there are a couple of dialogs that can be shown within the `Backpack` component